### ISSUE TITLE
Audio profiles not shown any more

### ISSUE THREAD

**Issue by silvervesi** [kgshank/gse-sound-output-device-chooser#165]
The profile selection is enabled but it does not no profiles are shown. Switching betweed HDP and A2DP works fine from Ubuntu 20.04's sound settings. Tried reinstalling too, no effect.

![image](https://user-images.githubusercontent.com/31201374/119143049-d8f43d80-ba4f-11eb-945a-9ff7f26b5d4d.png)

**Comment by vanhove** [comment]
I am also experiencing this issue on the same Ubuntu version.

**Comment by thesmallcreeper** [comment]
Can you try
1. Enabling new profile identification
2. Getting latest git version

**Comment by hannesvanhove-tomtom** [comment]
The issue still persists with the latest git version

**Comment by vanhove** [comment]
The issue still persists with the latest git version

**Comment by silvervesi** [comment]
Enabling new profile identification actually worked for me, but I had to disconnect the headphones and disable/enable the extension.

**Comment by thesmallcreeper** [comment]
@vanhove Using the latest git version, can you send:

1. A file with the journal messages of that session (`journalctl -r | grep SDC`)
2. The output of the `pa_helped.py` in a file? (locates in `/sound-output-device-chooser@kgshank.net/utils/pa_helped.py`, you should run it using `python`. In my case I had to run `python3 pa_helped.py`)

If so I will have a look at some point.

**Comment by tommie-lie** [comment]
I had the same problem and for me it was a layer 8 issue.
I could run `python3 pa_helped.py` but neither old nor new profile identification worked for me. Debugging the extension, I wondered why the output of the script never appeared in the JavaScript part.
The reason was that the extension executes `python` first, if it is available. I had previously installed the package `python-is-package3`, then set up pyenv for my development purposes and somehow managed to uninstall `python-is-python3`. This caused pyenv to still supply the `python` command, but not the system version of it (because system `python` didn't exist anymore) but a custom compiled python which, lacking a virtual environment, also lacked some extensions required by the sound device choser extension. Re-installing `python-is-python3` fixed it for me.

tl;dr: check if you maybe have broken your `python` command by accident, installing `python-is-python3` and setting pyenv's global version to "system" may help.

**Comment by thesmallcreeper** [comment]
@tommie-lie How can we improve the extension? Maybe calling `python3` and then `python` if not found?

**Comment by tommie-lie** [comment]
That won't help, because I could manage to configure `pyenv` so that `python3` is also the wrong Python.
For me, it was really a configuration mistake that could only have been resolved if the extension would have explicitly called `/usr/bin/python3` (circumventing pyenv). However, hardcoding `/usr/bin/python3` may not work everywhere, like systems that only have Python in `/usr/local/bin` (but I don't know if that is reasonable today, aside from maybe Linux from Scratch).

Considering that 1) Python 2.x has long been discontinued and 2) all major distributions have merged /bin and /usr/bin and 3) virtually all distributions ship with Python by default, it **may** be safe to just hard-code `#!/usr/bin/python3` in the helper script and just spawn the script directly (i.e. `GLib.spawn_command_line_sync(pyLocation);`. I guess that will work reasonably well, but I really don't know what other consequences there might be.
